### ISSUE TITLE
Homebridge directory not created after install?

### ISSUE THREAD

**Issue by ebpeterkin** [pdlove/homebridge-smartthings#201]
Hello,

I'm new to most of this but just trying to follow instructions to get HomeBridge-Smartthings installed on my RPI2 .  I'm following this article :

http://squidstack.com/smarthome/configuring-smartthings-to-homekit-with-homebridge/

I was able to get through the following commands successfully (I think, I didnt see any errors)

sudo apt-get update
sudo apt-get upgrade
sudo apt-get install git make

sudo apt-get install npm
sudo apt-get install libavahi-compat-libdnssd-dev

sudo npm cache clean -f
sudo npm install -g n
sudo n stable

sudo npm install -g homebridge --unsafe-perm
sudo npm install -g homebridge-smartthings

After these steps the next step is to change directory to the homebridge directory using this command:

cd /home/pi/.homebridge

But I keep getting :
bash: cd: /home/pi/.homebridge: No such file or directory

Does this mean the install was actually unsuccessful as the directory was not created? or maybe the path for the Homebridge directory has changed in a newer version created after this document was written? I have attached a screenshot of the response after running the homebridge install. Any help would be greatly appreciated.

![image](https://user-images.githubusercontent.com/30765727/29008729-d557bd4a-7ae9-11e7-845f-ba9b2d65e76a.png)

**Comment by pdlove** [comment]
You have to manually create the .homebridge folder and put your Config in it.

**Comment by pdlove** [comment]
For reference, here is the guide that I know works: https://blog.bram.co.nl/howto-homebridge-on-raspberry-pi-3-for-ios-and-smart-devices/
The only bit is to replace Homebridge-neatamo with the SmartThings items.

**Comment by ebpeterkin** [comment]
Thanks!!.. Just manually creating the directory got me much further along.
I've gotten to the point where homebridge seems to be running :


[image: Inline image 1]


But when I connect to the Home app and try to add an accessory Homebridge
does not show up. Interestingly, HomeBridge did show up earlier in the Home
app but I was not seeing any of the connected devices, when I looked closer
I noticed that I was gettign an access denied error when homebridge
started. This was caused by the incorrect  URL  which I changed and then
homebridge seemed to start fine. My problem now though a sI mentioned
before is that HomeBridge does not show up in the Home app when I try to
add an accessory. any ideas?




Regards,

Ellon B. Peterkin
Tel : *954-205-3256*
Fax: 561-948-8055
Skype/Hangouts: ebpeterkin

**Comment by ebpeterkin** [comment]
I see in other issues where you requested a dump. I followed those steps and below is a dump if that will help with troubleshooting this:

{"accessories":[{"aid":1,"services":[{"iid":1,"type":"0000003E-0000-1000-8000-0026BB765291","characteristics":[{"iid":2,"type":"00000014-0000-1000-8000-0026BB765291","perms":["pw"],"format":"bool","description":"Identify"},{"iid":3,"type":"00000020-0000-1000-8000-0026BB765291","perms":["pr"],"format":"string","value":"Default-Manufacturer","description":"Manufacturer"},{"iid":4,"type":"00000021-0000-1000-8000-0026BB765291","perms":["pr"],"format":"string","value":"Default-Model","description":"Model"},{"iid":5,"type":"00000023-0000-1000-8000-0026BB765291","perms":["pr"],"format":"string","value":"Homebridge","description":"Name"},{"iid":6,"type":"00000030-0000-1000-8000-0026BB765291","perms":["pr"],"format":"string","value":"Default-SerialNumber","description":"Serial Number"}]},{"iid":7,"type":"49FB9D4D-0FEA-4BF1-8FA6-E7B18AB86DCE","characteristics":[{"iid":8,"type":"77474A2F-FA98-485E-97BE-4762458774D8","perms":["pr","ev"],"format":"uint8","value":0,"description":"State","maxValue":1,"minValue":0,"minStep":1},{"iid":9,"type":"FD9FE4CC-D06F-4FFE-96C6-595D464E1026","perms":["pr","ev"],"format":"string","value":"1.0","description":"Version"},{"iid":10,"type":"5819A4C2-E1B0-4C9D-B761-3EB1AFF43073","perms":["pr","pw","ev"],"format":"data","value":null,"description":"Control Point"}]}]},{"aid":2,"services":[{"iid":1,"type":"0000003E-0000-1000-8000-0026BB765291","characteristics":[{"iid":2,"type":"00000014-0000-1000-8000-0026BB765291","perms":["pw"],"format":"bool","description":"Identify"},{"iid":3,"type":"00000020-0000-1000-8000-0026BB765291","perms":["pr"],"format":"string","value":"Default-Manufacturer","description":"Manufacturer"},{"iid":4,"type":"00000021-0000-1000-8000-0026BB765291","perms":["pr"],"format":"string","value":"Default-Model","description":"Model"},{"iid":5,"type":"00000023-0000-1000-8000-0026BB765291","perms":["pr"],"format":"string","value":"PLAYBAR","description":"Name"},{"iid":6,"type":"00000030-0000-1000-8000-0026BB765291","perms":["pr"],"format":"string","value":"Default-SerialNumber","description":"Serial Number"}]},{"iid":7,"type":"00000049-0000-1000-8000-0026BB765291","characteristics":[{"iid":8,"type":"00000025-0000-1000-8000-0026BB765291","perms":["pr","pw","ev"],"format":"bool","value":false,"description":"On"}]}]}]}

**Comment by ebpeterkin** [comment]
Disregard. I resolved the issue. Apparently both wlan0 and eth0 are active on my RPI . I was configuring Homebridge for wlan0 but  it's actually using eth0. I assumed it was using wlan0 because of a message I saw about DirectConnect listening on 192.168.55.215. Seems that is Smartthings though , so it seems to be using wlan0 for the Smartthings side of the bridge and eth0 for the Homekit side of the bridge?  Anyway , I'm just glad its working , thanks for your help!
